src/libraries/System.Resources.Extensions/tests/BinaryFormatTests/Common/BasicObjectTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Runtime.Serialization.Formatters;
+using BinaryFormatTests;
+using BinaryFormatTests.FormatterTests;
+
+namespace System.Resources.Extensions.Tests.Common;
+
+public abstract class BasicObjectTests<T> : SerializationTest<T> where T : ISerializer
+{
+    private protected abstract bool SkipOffsetArrays { get; }
+
+    [Theory]
+    [MemberData(nameof(SerializableObjects))]
+    public void DeserializeStoredObjects(object value, TypeSerializableValue[] serializedData)
+    {
+        _ = value;
+
+        int platformIndex = serializedData.GetPlatformIndex();
+        for (int i = 0; i < serializedData.Length; i++)
+        {
+            for (FormatterAssemblyStyle assemblyMatching = 0; assemblyMatching <= FormatterAssemblyStyle.Full; assemblyMatching++)
+            {
+                object deserialized = DeserializeFromBase64Chars(serializedData[i].Base64Blob, assemblyMatching: assemblyMatching);
+
+                if (deserialized is StringComparer)
+                {
+                    // StringComparer derived classes are not public and they don't serialize the actual type.
+                    value.Should().BeAssignableTo<StringComparer>();
+                }
+                else
+                {
+                    deserialized.Should().BeOfType(value.GetType());
+                }
+
+                bool isSamePlatform = i == platformIndex;
+                EqualityExtensions.CheckEquals(value, deserialized, isSamePlatform);
+            }
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(BasicObjectsRoundtrip_MemberData))]
+    public void BasicObjectsRoundtrip(
+        object value,
+        FormatterAssemblyStyle assemblyMatching,
+        FormatterTypeStyle typeStyle)
+    {
+        object deserialized = RoundTrip(value, typeStyle: typeStyle, assemblyMatching: assemblyMatching);
+
+        // string.Empty and DBNull are both singletons
+        if (!ReferenceEquals(value, string.Empty)
+            && value is not DBNull
+            && value is Array array
+            && array.Length > 0)
+        {
+            deserialized.Should().NotBeSameAs(value);
+        }
+
+        EqualityExtensions.CheckEquals(value, deserialized, isSamePlatform: true);
+    }
+
+    public static EnumerableTupleTheoryData<object, TypeSerializableValue[]> SerializableObjects()
+    {
+        // Can add a .Skip() to get to the failing scenario easier when debugging.
+        return new((
```

Comment:
More instances of return-target-typed-new here; and probably elsewhere in the PR.  I won't point out any other ones this round.

Reply:
Do you really see the value in specifying the full (and very long) name of the type that is visible just few lines above?

https://github.com/dotnet/runtime/blob/b57bc2af4714d73ca759e9d7c5bc65c1293236b3/src/libraries/System.Resources.Extensions/tests/BinaryFormatTests/Common/BasicObjectTests.cs#L65-L68

https://github.com/dotnet/runtime/blob/b57bc2af4714d73ca759e9d7c5bc65c1293236b3/src/libraries/System.Resources.Extensions/tests/BinaryFormatTests/Common/BasicObjectTests.cs#L75-L77

I am not convinced that it would add any value.

Reply:
It doesn't really matter what I think here, or what you think here.  We have a style guideline.  Until/unless you can get it changed, you should follow it.